dlp/src/test/java/com/example/dlp/DeIdentificationIT.java

```diff
@@ -56,26 +56,49 @@ public void setUp() {
   @Test
   public void testDeidStringMasksCharacters() throws Exception {
     String text = "\"My SSN is 372819127\"";
-    DeIdentification.main(new String[] {
-        "-m", text,
-        "-maskingCharacter", "x",
-        "-numberToMask", "5"
-    });
+    DeIdentification.main(
+        new String[] {
+          "-m", text,
+          "-maskingCharacter", "x",
+          "-numberToMask", "5"
+        });
     String output = bout.toString();
-    assertEquals(output, "My SSN is xxxxx9127\n");
+    assertThat(output, containsString("My SSN is xxxxx9127"));
   }
 
   @Test
   public void testDeidStringPerformsFpe() throws Exception {
     String text = "\"My SSN is 372819127\"";
-    DeIdentification.main(new String[] {
-        "-f", text,
-        "-wrappedKey", wrappedKey,
-        "-keyName", keyName
-    });
+    DeIdentification.main(
+        new String[] {
+          "-f", text,
+          "-wrappedKey", wrappedKey,
+          "-keyName", keyName
+        });
+    String output = bout.toString();
+    assertFalse(
+        "Response contains original SSN.",
+        output.contains("372819127"));
+    assertThat(output, containsString("My SSN is "));
+  }
+
+  @Test
+  public void testDeidentifyWithDateShift() throws Exception {
+    DeIdentification.main(
+        new String[] {
+            "-d",
+            "-inputCsvPath", "src/test/resources/dates.csv",
+            "-outputCsvPath", "src/test/resources/results.temp.csv",
+            "-dateFields", "birth_date,register_date",
+            "-lowerBoundDays", "5",
+            "-upperBoundDays", "5",
+            "-contextField", "name",
+            "-wrappedKey", wrappedKey,
+            "-keyName", keyName
+        });
     String output = bout.toString();
-    assertFalse(output.contains(text));
-    assertTrue(Pattern.compile("My SSN is \\w+").matcher(output).find());
+    assertThat(
+        output, containsString("Successfully saved date-shift output to: results.temp.csv"));
```

Comment:
Should we also check the contents of `results.temp.csv` against a file containing the desired results?